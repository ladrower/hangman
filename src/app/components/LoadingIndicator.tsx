import { observe } from '@/infrastructure/store'
import React from 'react'

type Props = {
  resource: { loading?: boolean }
  component: React.ComponentType
}

export function LoadingIndicator(
  {
    resource,
    component: Component
  }: Props) {
  return observe(resource, ({loading }) => loading ? <Component /> : null)
}
