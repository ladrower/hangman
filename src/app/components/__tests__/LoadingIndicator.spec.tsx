import React from 'react';
import renderer from 'react-test-renderer';
import { LoadingIndicator } from '../LoadingIndicator'

const Spinner = () => <div>spinner...</div>

it('should render spinner', () => {
    const tree = renderer
        .create(<LoadingIndicator resource={{ loading: true }} component={Spinner}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should not render spinner', () => {
    const tree = renderer
        .create(<LoadingIndicator resource={{ loading: false }} component={Spinner}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
