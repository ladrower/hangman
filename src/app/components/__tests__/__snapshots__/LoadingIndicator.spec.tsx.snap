// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render spinner 1`] = `null`;

exports[`should render spinner 1`] = `
<div>
  spinner...
</div>
`;
